sources/osg/Program.js

```diff
@@ -6,15 +6,23 @@ var StateAttribute = require( 'osg/StateAttribute' );
 var CustomMap = require( 'osg/Map' );
 var Timer = require( 'osg/Timer' );
 
+
+/*develblock:start*/
+var defineShaderNameReg = /#define\sSHADER_NAME\s(\S+)/im;
+var unnamedProgrammNumber = 0;
+/*develblock:end*/
+
 /**
  * Program encapsulate an vertex and fragment shader
  * @class Program
  */
-var Program = function ( vShader, fShader ) {
+var Program = function ( vShader, fShader, name, doTiming ) {
```

Comment:
dont change the signature, name can be called with setName. doTiming can be computed at runtime that could depends on log level info.
Can check Notify level if it's <= info you can display timing. It will be display for debug and info level